### DISCUSSION TITLE
CSP primefaces 8 block jquery script

### DISCUSSION THREAD

**Discussion by Yeison07** [primefaces/community#2867] [PrimeFaces] [answered]
Hi I need help to correctly configure my CSP with primefaces 8, I have tried creating a custom filter, custimizing in the web.xml with.
<param-name>primefaces.CSP_POLICY</param-name>
But with each attempt I am blocked with a different script, currently with this very basic configuration the primefaces jquery is blocked, if you can guide me with the process or any documentation link I would appreciate it.

**Web.xml**
```
<context-param>
    <param-name>primefaces.CSP</param-name>
    <param-value>true</param-value>
  </context-param>

```
**Console.log**
`Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'nonce-MzAwOGNhMjYtYTFjNS00ZWEzLWFmYjMtZDhjYTEyMzNiZjRm'". Either the 'unsafe-inline' keyword, a hash ('sha256-8YLY9DtRiNRagr+gfqiWweqzA8zoHmxtprZb1pxgPO4='), or a nonce ('nonce-...') is required to enable inline execution.
jquery.js?ln=primefaces&v=8.0:2
`
**Login.xhtml button**
```
<div class="group">
<p:commandButton value="Ingresar" id="uno"
update="form:msg"
process="@this :form"
action="#{loginFormBean.autenticar}"
styleClass="button"
oncomplete="btn_desactivar()" />
</div>
```

**Comment by melloware** [comment]
You shouldn't have to do anything.  PrimeFaces rearranges the scripts to work can you put together a PrimeFaces Test reproducer of your issue?

**Reply by Yeison07** [reply]
In this repository I leave minimum reproducible example, thank you very much for your help 👋
https://github.com/Yeison07/primefaces-csp-issue

**Reply by melloware** [reply, marked as answer]
@Yeison07 your code is invalid.  Your login.xhtml inherits from `template.xhtml` which is not correct only 1 template should have an `<html><head><body>` you need to do includes properly.

I fixed your example and now its not getting the CSP error.

[primefaces-csp-issue-master.zip](https://github.com/user-attachments/files/17395283/primefaces-csp-issue-master.zip)

**Reply by Yeison07** [reply]
That was indeed the problem, thank you very much for your help and your time.
